manifests/base/argo-server/argo-server-deployment.yaml

```diff
@@ -26,5 +26,8 @@ spec:
               path: /
             initialDelaySeconds: 10
             periodSeconds: 20
+      securityContext:
+        runAsNonRoot: true
```

Comment:
i'm pretty sure runAsNonRoot is unnecessary when you have runAsUser. 

I think most peoples preference (including mine) seem to be to set a numeric userid in the Dockerfile and only having `runAsNonRoot: true` in the manifests, and omit `runAsUser` from the manifests. 

Could we do that instead? It would be consistent with CD and Rollouts are doing.